yaml/metadata.go

```diff
@@ -23,8 +33,45 @@ func (m *Metadata) ToPipeline() *pipeline.Metadata {
 	} else {
 		clone = *m.Clone
 	}
+
 	return &pipeline.Metadata{
-		Template: m.Template,
-		Clone:    clone,
+		Template:    m.Template,
+		Clone:       clone,
+		Environment: m.Environment,
 	}
 }
+
+// HasEnvironment checks if the container type
+// is contained within the environment list.
+func (m *Metadata) HasEnvironment(container string) bool {
+	for _, e := range m.Environment {
+		if e == container {
+			return true
+		}
+	}
+
+	return false
+}
+
+// UnmarshalYAML implements the Unmarshaler interface for the Metadata type.
+func (m *Metadata) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	// metadata we try unmarshalling to
+	metadata := new(_metadata)
```

Comment:
Related to my earlier comment - we could declare an anonymous struct here instead.

This would follow the conventions and patterns already established for other types.

Specifically where we override the YAML unmarshaling behavior:

https://github.com/go-vela/types/blob/d9de354942deb1a8104aa1ba451b88e342555199/yaml/ruleset.go#L56-L63

https://github.com/go-vela/types/blob/d9de354942deb1a8104aa1ba451b88e342555199/yaml/ruleset.go#L125-L134

Reply:
Belive this is what you were looking for:
https://github.com/go-vela/types/pull/186/commits/cac25f3ed967c1b8286db45541162783b347f580